ckan-2.10/base/Dockerfile.py-3.10

```
@@ -0,0 +1,92 @@
+# use the official Python 3.10 (bookworm slim) image as the base image
+FROM python:3.10-slim-bookworm
+
+# Tag passed through via the Makefile
+ARG CKAN_TAG=${CKAN_TAG}
+
+ENV TZ=UTC
+ENV APP_DIR=/srv/app
+ENV SRC_DIR=${APP_DIR}src
+ENV CKAN_INI=${APP_DIR}/ckan.ini
+ENV PIP_SRC=${SRC_DIR}
+ENV CKAN_STORAGE_PATH=/var/lib/ckan
+ENV GIT_URL=https://github.com/ckan/ckan.git
+
+# Customize these in the environment (.env) file if needed
+ENV CKAN_SITE_URL=http://localhost:5000
+ENV CKAN__PLUGINS image_view text_view recline_view datastore envvars
+
+# UWSGI options
+ENV UWSGI_HARAKIRI=50
+
+WORKDIR ${APP_DIR}
+
+# Set up timezone
+RUN echo ${TZ} > /etc/timezone
+
+# Set LC_ALL=en_US.UTF-8 will ensure that all locale-dependent operations in the current environment 
+# will use English language and United States cultural conventions with UTF-8 character encoding
+ENV LC_ALL=en_US.UTF-8
+
+# Set the locale          
+RUN apt-get update
+RUN apt-get install --no-install-recommends -y locales
+RUN sed -i "/$LC_ALL/s/^# //g" /etc/locale.gen
+RUN dpkg-reconfigure --frontend=noninteractive locales 
+RUN update-locale LANG=${LC_ALL}
+
+# Install system libraries
+RUN apt-get install --no-install-recommends -y \
+        apt-utils \
+        git \
+        libpq-dev \
+        g++ \
+        linux-headers-generic \
+        libtool \
+        wget
+
+# Create the src directory
+RUN mkdir -p ${SRC_DIR}
+
+# Install supervisord and create the supervisord.d directory
+RUN apt-get install --no-install-recommends -y \
+        supervisor && \
+        mkdir /etc/supervisord.d
+
+COPY setup/supervisord.conf /etc
+
+# Install uwsgi, the CKAN application, the dependency packages for CKAN plus some confiquration
+RUN python3 -m venv ${APP_DIR} && \
```

Can we create the venv in a separate location (/srv/app/venv) to keep things tidy

```suggestion
RUN python3 -m venv ${VENV_DIR} && \
```

---

More generally I wonder if [`/srv/app`](https://tldp.org/LDP/Linux-Filesystem-Hierarchy/html/srv.html) was a good historic choice for locating the ckan files, perhaps [`/opt/ckan`](https://tldp.org/LDP/Linux-Filesystem-Hierarchy/html/opt.html) would have made more sense.
In any case I think that now it's a bit late to change that. WDYT @kowh-ai @wardi ?

---

I'm not a FHS expert but I think /srv/ is fine for files we're installing ourselves outside the package manager's control